### PR TITLE
added self_link to hvn, peering and tgw attachment

### PR DESCRIPTION
**Pull request by smaant**
### :hammer_and_wrench: Description

Added `self_link` attribute to hvn, peering and tgw attachment resources and datasources.

### CODE REVIEW

**Review by nombiezinja — COMMENTED** [review]
LGTM but am unfamiliar with the repo so will let Brenna do the approval!

**Review by bcmdarroch — COMMENTED** [review]
Looks good! My only ask: could you update the acceptance tests too? https://github.com/hashicorp/terraform-provider-hcp/blob/main/internal/provider/resource_hvn_test.go

**Comment by smaant** [comment]
@bcmdarroch good point, thank you for noticing! updated the test

File: internal/provider/resource_hvn_test.go [hunk 1]
```diff
@@ -102,6 +106,41 @@ func testAccCheckHvnExists(name string) resource.TestCheckFunc {
 	}
 }
 
+func testSelfLink(name string, expectedID, expectedType string) resource.TestCheckFunc {
```
Comment (bcmdarroch) [inline internal/provider/resource_hvn_test.go:109]: thanks for adding this! probably something we could share with future tests :)

Comment (smaant) [reply internal/provider/resource_hvn_test.go:109]: yes, this is how I saw it too

**Review by bcmdarroch — APPROVED** [review]

**Pull request merged**